project-base/storefront/components/Pages/Customer/customerChangeProfileFormMeta.ts

```diff
@@ -29,14 +29,15 @@ export const useCustomerChangeProfileForm = (
     const resolver = yupResolver(
         Yup.object().shape<Record<keyof CustomerChangeProfileFormType, any>>({
             email: validateEmail(t),
+            passwordOld: Yup.string(),
             passwordFirst: Yup.string().when('passwordOld', {
                 is: (passwordOld: string) => passwordOld.length > 0,
-                then: validateFirstPassword(t),
+                then: validatePassword(t),
                 otherwise: Yup.string(),
             }),
             passwordSecond: Yup.string().when('passwordFirst', {
                 is: (passwordFirst: string) => passwordFirst.length > 0,
-                then: validateSecondPassword(t),
+                then: validateRepeatedPassword(t),
```

Comment:
There are quite some discrepancies in naming of all these password variables and validation rules. This is true on multiple levels:
- one name is given by the API (e.g. we have `newPassword`)
- another name is given by the form definition (e.g. we have `passwordFirst`)
- and sometimes another name is also given by the name of the validation rule (e.g. we have `passwordSecond` vs `validateRepeatedPassword`)

The best approach IMO would be to follow API naming as much as possible and as much as it makes sense (of course, API does not have any repeated passwords, but that should be easy to solve). You can find the name of all API argument names in the YAML definitions for a given mutation (e.g. `ChangePasswordInputDecorator`)